e2e/stack_test.go

```diff
@@ -117,6 +122,46 @@ func TestE2E(t *testing.T) {
 	runningTests.Wait()
 }
 
+func checkForRollbacks(t *testing.T, stack *e2e.StackConfig) {
+	// Subscribe to new block header events
+	const subscriber = "rollbackChecker"
+	eventChan, err := stack.L2Client.Subscribe(stack.Ctx, subscriber, bfttypes.QueryForEvent(bfttypes.EventNewBlockHeader).String())
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, stack.L2Client.UnsubscribeAll(stack.Ctx, subscriber))
+	}()
+
+	l1BlockTicker := time.NewTicker(250 * time.Millisecond)
+	defer l1BlockTicker.Stop()
+
+	var lastBlockHeight int64
+
+	for {
+		select {
+		// Check new block header events for rollbacks
+		case event := <-eventChan:
+			eventNewBlockHeader, ok := event.Data.(bfttypes.EventDataNewBlockHeader)
+			require.True(t, ok)
+			currentHeight := eventNewBlockHeader.Header.Height
+
+			// Ensure that the current block height is not less than the last checked height
+			if lastBlockHeight > 0 && currentHeight < lastBlockHeight {
+				require.FailNow(t, "monomer has rolled back")
```

Comment:
nit: I think we can use slightly more idiomatic `require`s:

```golang
require.Equal(t, currentHeight, lastBlockHeight+1)
```

If I'm understanding correctly, we may still need to guard this by an `if` statement to properly initialize `lastBlockHeight`:

```golang
if lastBlockHeight > 0 {
    require.Equal(t, currentHeight, lastBlockHeight+1)
}
```

If that understanding is correct, it would be good to add a comment above the conditional explaining why we need `lastBlockHeight > 0`.

Reply:
Good idea, added https://github.com/polymerdao/monomer/pull/192/commits/dd88c16a87206cc8cc9d9d14a6d3951e48913ae8